eer support

azazellochg opened the pull request:
Mc2 does not identify defects as in relion mc, so a separate defect file is required.

delarosatrevin reviewed (commented):
Some comments

azazellochg commented on motioncorr/convert.py:
```diff
@@ -24,6 +24,186 @@
+from tifffile import TiffFile
+
+
+class OpticsGroups:
```
it is duplicated, yes. Do you think it's a good idea to add another dep?

delarosatrevin replied:
Hum...not sure what might be the best, if duplicate the whole piece of code or add a dep. Other mc protocols might also use the OpticsGroup stuff?

azazellochg replied:
There are no other plugins afaik that support eer.

azazellochg replied:
How about we put this class into emtable?

azazellochg commented on motioncorr/convert.py:
```diff
@@ -58,3 +238,43 @@ def getMovieFileName(movie):
     return fn
+
+
+def parseEERDefects(fn):
```
So far, relion extracts defects from the gain file differently - 0 pixels mean a defect. Here I extract the same info from the header. It might change in the future (as always with TFS). But I think any plugin that does motion cor on EPU EER data will potentially benefit from this.

azazellochg commented on motioncorr/protocols/protocol_motioncorr.py:
```diff
@@ -377,6 +448,19 @@ def _validate(self):
+            movie = firstMovie.getFileName()
+            imgx, imgy, _, _ = ImageHandler().getDimensions(movie)
+
+            if sorted([gainx, gainy]) != sorted([imgx, imgy]):
```
e.g. if the gain is flipped it could be x, y vs y, x

azazellochg commented on motioncorr/protocols/protocol_motioncorr.py:
```diff
@@ -377,6 +448,19 @@ def _validate(self):
 
+        # check gain dimensions and extension
+        if inputMovies.getGain():
+            from pwem.emlib.image import ImageHandler
```
Will do, thanks!

azazellochg commented:
I'll run a test this week to check if polishing works after mc2 on apoF data.

azazellochg commented:
Works okay!

azazellochg commented:
@delarosatrevin I have added relino requirement, fixed the frame numbers for eer and also now save shifts relative to the first frame. Originally mc2 always did that but now the output log shows shifts depending on FmRef (default is middle frame)
